asmcomp/amd64/proc.ml

```diff
@@ -325,6 +325,7 @@ let destroyed_at_oper = function
   | Iop (Iintop_imm(Icheckbound _, _)) when Config.spacetime ->
       [| loc_spacetime_node_hole |]
   | Iswitch(_, _) -> [| rax; rdx |]
+  | Itrywith _ -> [| r11 |]
```

Comment:
Can you say a few words about this change?

Reply:
The `trywith` needs an auxiliary register to load things from memory and push them on stack. Previously it would first push `r14` (the location of the previous trap handler) and then use `r14` as a temporary register (clever!) before setting it to the new trap-handler value (the current value of `rsp`), so it appeared to not use any extra register. With the new register usage scheme, `r14` is the domain state pointer and cannot be used in this way, so a new auxiliary register is needed. It looks like r11 was used in a rather arbitrary way, possibly because it is already used by `raise` in a sort of similar fashion.

Reply:
@gasche has explained it well. `r11` was chosen among the available registers because it is the last general purpose register that is picked by the register allocator. Moreover, `r11` is not used for OCaml argument passing and is destroyed at c calls and plt stubs (allocations in particular). Hence, it is less likely that the register is alive at a `try_with` compared to say `rax`.

Reply:
Got it, thanks; I will need to rebase my risc-v backend on top of this patch so the explanation is helpful!